drivers/scheduler/k8s/k8s.go

```diff
@@ -5231,6 +5233,59 @@ func (k *K8s) CSICloneTest(ctx *scheduler.Context, request scheduler.CSICloneReq
 	return nil
 }
 
+// CSISnapshotAndRestoreMany create CSI snapshot and restore to many PVCs, and we validate the PVCs are up and bound
+// (Testrail cases C58775, 58091)
+func (k *K8s) CSISnapshotAndRestoreMany(ctx *scheduler.Context, request scheduler.CSISnapshotRequest) error {
+	// This test will validate the content of the volume as opposed to just verify creation of volume.
+	if !k.PureRunCloneMany {
+		logrus.Info("PureRunCloneMany job disabled, skipping")
+		return nil
+	}
+	pvcObj, err := k8sCore.GetPersistentVolumeClaim(request.OriginalPVCName, request.Namespace)
+	size := pvcObj.Spec.Resources.Requests[corev1.ResourceStorage]
+
+	if err != nil {
+		logrus.Errorf("Failed to retrieve PVC %s in namespace: %s : %s", request.OriginalPVCName, request.Namespace, err)
+		return err
+	}
+	originalStorageClass, err := k8sCore.GetStorageClassForPVC(pvcObj)
+	if err != nil {
+		logrus.Errorf("Failed to retrieve SC for PVC %s in namespace: %s : %s", request.OriginalPVCName, request.Namespace, err)
+		return err
+	}
+	storageClassName := originalStorageClass.Name
+	logrus.Infof("Procedding with SC %s", storageClassName)
+
+	// creating the snapshot
+	volSnapshot, err := k.CreateCsiSnapshot(request.SnapName, pvcObj.Namespace, request.SnapshotclassName, pvcObj.Name)
+	if err != nil {
+		logrus.Errorf("Failed to create snapshot %s for volume %s", request.SnapName, pvcObj.Name)
+		return err
+	}
+
+	logrus.Infof("Successfully created snapshot: [%s] for pvc: %s", volSnapshot.Name, pvcObj.Name)
+	for i := 0; i < 500; i++ {
+		restoredPVCName := fmt.Sprint(request.RestoredPVCName, i)
+		restoredPVCSpec, err := GeneratePVCRestoreSpec(size, pvcObj.Namespace, restoredPVCName, volSnapshot.Name, storageClassName)
+		if err != nil {
+			logrus.Errorf("Failed to build cloned PVC Spec: %s", err)
+			return err
+		}
+		_, err = k8sCore.CreatePersistentVolumeClaim(restoredPVCSpec)
+		if err != nil {
+			logrus.Errorf("Failed to restore PVC from snapshot %s: %s", volSnapshot.Name, err )
+			return err
+		}
+
+		if err = k.waitForPVCToBound(restoredPVCName, pvcObj.Namespace); err != nil {
```

Comment:
Let's avoid waiting here. Use the for loop to send requests parallel and have another for loop to check PVCs bound status later.